File: app/src/main/java/com/example/wpossbank/MainActivity.java
package com.example.wpossbank;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.example.wpossbank.modelos.User;

import eightbitlab.com.blurview.BlurView;
import eightbitlab.com.blurview.RenderScriptBlur;

public class MainActivity extends AppCompatActivity{
    Context context;
    User user;

    BlurView blurView;

    TextView userView;
    Button cardPaymentsButton, withdrawalsButton, depositsButton,
            transfersButton, balanceButton, historyButton, logOutButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        context = this;
        user = new User(context);

        blurView = findViewById(R.id.blurView);
        blurBackground();

        userView = findViewById(R.id.userView);

        cardPaymentsButton = findViewById(R.id.cardPaymentsButton);
        withdrawalsButton = findViewById(R.id.withdrawalsButton);
        depositsButton = findViewById(R.id.depositsButton);
        transfersButton = findViewById(R.id.transfersButton);
        balanceButton = findViewById(R.id.balanceButton);
        historyButton = findViewById(R.id.historyButton);
        logOutButton = findViewById(R.id.logOutButton);

        user.loadData(user);
        userView.setText(getString(R.string.main_welcome_message, user.getName()));

        cardPaymentsButton.setOnClickListener(showCardPayment ->
            startActivity(new Intent(context, CardPaymentActivity.class)));

        withdrawalsButton.setOnClickListener(showWithdraws ->
            startActivity(new Intent(context, WithdrawalsActivity.class)));

        depositsButton.setOnClickListener(showDeposits ->
            startActivity(new Intent(context, DepositsActivity.class)));

        transfersButton.setOnClickListener(showTransfers ->
            startActivity(new Intent(context, TransfersActivity.class)));

        balanceButton.setOnClickListener(showBalance ->
            startActivity(new Intent(context, GetBalanceActivity.class)));

        historyButton.setOnClickListener(showHistory ->
                startActivity(new Intent(context, TransfersHistoryActivity.class)));

        logOutButton.setOnClickListener(finishActivity -> finish());
    }

    private void blurBackground(){
        float radius = 20f;

        View decorView = getWindow().getDecorView();
        ViewGroup rootView = decorView.findViewById(android.R.id.content);
        Drawable windowBackground = decorView.getBackground();

        blurView.setupWith(rootView)
                .setFrameClearDrawable(windowBackground)
                .setBlurAlgorithm(new RenderScriptBlur(this))
                .setBlurRadius(radius)
                .setBlurAutoUpdate(true)
                .setHasFixedTransformationMatrix(true);
    }
}

File: app/src/main/java/com/example/wpossbank/activities/MainActivity.kt
package com.example.wpossbank.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.wpossbank.R
import com.example.wpossbank.models.User
import eightbitlab.com.blurview.BlurView
import eightbitlab.com.blurview.RenderScriptBlur

class MainActivity : AppCompatActivity() {
    lateinit var context: Context
    lateinit var user: User
    lateinit var blurView: BlurView
    private lateinit var userView: TextView
    private lateinit var cardPaymentsButton: Button
    private lateinit var withdrawalsButton: Button
    private lateinit var depositsButton: Button
    private lateinit var transfersButton: Button
    private lateinit var balanceButton: Button
    private lateinit var historyButton: Button
    private lateinit var logOutButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        context = this
        user = User()
        blurView = findViewById(R.id.blurView)
        blurBackground()
        userView = findViewById(R.id.userView)
        cardPaymentsButton = findViewById(R.id.cardPaymentsButton)
        withdrawalsButton = findViewById(R.id.withdrawalsButton)
        depositsButton = findViewById(R.id.depositsButton)
        transfersButton = findViewById(R.id.transfersButton)
        balanceButton = findViewById(R.id.balanceButton)
        historyButton = findViewById(R.id.historyButton)
        logOutButton = findViewById(R.id.logOutButton)
        user.loadData(context)
        userView.text = getString(
            R.string.main_welcome_message,
                user.name.split(" ").toTypedArray()[0])

        cardPaymentsButton.setOnClickListener {
            startActivity(
                Intent(context, CardPaymentActivity::class.java)
            )
        }
        withdrawalsButton.setOnClickListener {
            startActivity(
                Intent(context, WithdrawalsActivity::class.java)
            )
        }
        depositsButton.setOnClickListener {
            startActivity(
                Intent(context, DepositsActivity::class.java)
            )
        }
        transfersButton.setOnClickListener {
            startActivity(
                Intent(context, TransfersActivity::class.java)
            )
        }
        balanceButton.setOnClickListener {
            startActivity(
                Intent(context, GetBalanceActivity::class.java)
            )
        }
        historyButton.setOnClickListener {
            startActivity(
                Intent(context, TransfersHistoryActivity::class.java)
            )
        }
        logOutButton.setOnClickListener { finish() }
    }

    private fun blurBackground() {
        val radius = 20f
        val decorView = window.decorView
        val rootView = decorView.findViewById<ViewGroup>(android.R.id.content)
        val windowBackground = decorView.background
        blurView.setupWith(rootView)
            .setFrameClearDrawable(windowBackground)
            .setBlurAlgorithm(RenderScriptBlur(this))
            .setBlurRadius(radius)
            .setBlurAutoUpdate(true)
            .setHasFixedTransformationMatrix(true)
    }
}